### ISSUE TITLE
WLS10.3.6 - Unable to set attributes on WebAppContainer MBean

### ISSUE THREAD

**Issue by 53d117460ec63d70** [oracle/weblogic-deploy-tooling#324]
Model extract:
```yaml
topology:
    WebAppContainer:
        XPoweredByHeaderLevel: NONE
```
Create domain fails with the message:
```
<WLSDPLY-05026> <WebAppContainer is not one of the folder names allowed in model location topology:/>'
```

**Comment by rakillen** [comment]
WebAppContainer should be under resources, such as:
'''
resources:
     WebAppContainer:
         GzipCompression:
             GzipCompressionEnabled: true
             GzipCompressionMinContentLength: 5
             GzipCompressionContentType: [ text/html, text/xml ]
             Notes: "Compression info"
         AllowAllRoles: true
         Notes: "Container notes"
         WorkContextPropagationEnabled: true
         XPoweredByHeaderLevel: FULL
'''

**Comment by 53d117460ec63d70** [comment]
If I pass your model (given in your previous comment, unaltered) to domain create I get the same validation error. Are you able to reproduce this? I'm using WDT version 0.20 and WLS10.3.6:
```
$ cat /home/rob/build/mydomain.yml
resources:
     WebAppContainer:
         GzipCompression:
             GzipCompressionEnabled: true
             GzipCompressionMinContentLength: 5
             GzipCompressionContentType: [ text/html, text/xml ]
             Notes: "Compression info"
         AllowAllRoles: true
         Notes: "Container notes"
         WorkContextPropagationEnabled: true
         XPoweredByHeaderLevel: FULL
$ /home/rob/build/weblogic-deploy/bin/createDomain.sh -model_file /home/rob/build/mydomain.yml -oracle_home /home/rob/oracle_home -domain_home /home/rob/domain_home -domain_type WLS
JAVA_HOME = /home/rob/java_home
WLST_EXT_CLASSPATH =
CLASSPATH = /home/rob/build/weblogic-deploy/lib/weblogic-deploy-core.jar
WLST_PROPERTIES = -Djava.util.logging.config.class=oracle.weblogic.deploy.logging.WLSDeployLoggingConfig -Dcom.oracle.cie.script.throwException=true
/home/rob/oracle_home/wlserver_10.3/common/bin/wlst.sh /home/rob/build/weblogic-deploy/lib/python/create.py -oracle_home /home/rob/oracle_home -domain_home /home/rob/domain_home -domain_type WLS -model_file /home/rob/build/mydomain.yml -domain_type WLS

CLASSPATH=/home/rob/oracle_home/patch_wls1036/profiles/default/sys_manifest_classpath/weblogic_patch.jar:/home/rob/java_home/lib/tools.jar:/home/rob/oracle_home/wlserver_10.3/server/lib/weblogic_sp.jar:/home/rob/oracle_home/wlserver_10.3/server/lib/weblogic.jar:/home/rob/oracle_home/modules/features/weblogic.server.modules_10.3.6.0.jar:/home/rob/oracle_home/wlserver_10.3/server/lib/webservices.jar:/home/rob/oracle_home/modules/org.apache.ant_1.7.1/lib/ant-all.jar:/home/rob/oracle_home/modules/net.sf.antcontrib_1.1.0.0_1-0b2/lib/ant-contrib.jar:/home/rob/build/weblogic-deploy/lib/weblogic-deploy-core.jar:/home/rob/oracle_home/utils/config/10.3/config-launch.jar::/home/rob/oracle_home/wlserver_10.3/common/derby/lib/derbynet.jar:/home/rob/oracle_home/wlserver_10.3/common/derby/lib/derbyclient.jar:/home/rob/oracle_home/wlserver_10.3/common/derby/lib/derbytools.jar::

Initializing WebLogic Scripting Tool (WLST) ...

Welcome to WebLogic Server Administration Scripting Shell

Type help() for help on available commands

####<Mar 28, 2019 9:47:52 AM> <INFO> <Validator> <__validate_model_file> <WLSDPLY-05002> <Performing validation in TOOL mode for WebLogic Server version 10.3.6.0 and WLST OFFLINE mode>
####<Mar 28, 2019 9:47:52 AM> <INFO> <Validator> <__validate_model_file> <WLSDPLY-05003> <Performing model validation on the /home/rob/build/mydomain.yml model file>
####<Mar 28, 2019 9:47:52 AM> <INFO> <Validator> <__validate_domain_info_section> <WLSDPLY-05008> <Validating the domainInfo section of the model file>
####<Mar 28, 2019 9:47:52 AM> <INFO> <Validator> <__validate_domain_info_section> <WLSDPLY-05009> <Model file /home/rob/build/mydomain.yml does not contain a domainInfo section, validation of domainInfo was skipped.>
####<Mar 28, 2019 9:47:52 AM> <INFO> <Validator> <__validate_model_section> <WLSDPLY-05008> <Validating the topology section of the model file>
####<Mar 28, 2019 9:47:52 AM> <INFO> <Validator> <__validate_model_section> <WLSDPLY-05009> <Model file /home/rob/build/mydomain.yml does not contain a topology section, validation of topology was skipped.>
####<Mar 28, 2019 9:47:52 AM> <INFO> <Validator> <__validate_model_section> <WLSDPLY-05008> <Validating the resources section of the model file>
####<Mar 28, 2019 9:47:53 AM> <INFO> <Validator> <__validate_model_section> <WLSDPLY-05008> <Validating the appDeployments section of the model file>
####<Mar 28, 2019 9:47:53 AM> <INFO> <Validator> <__validate_model_section> <WLSDPLY-05009> <Model file /home/rob/build/mydomain.yml does not contain a appDeployments section, validation of appDeployments was skipped.>
####<Mar 28, 2019 9:47:53 AM> <SEVERE> <ValidationResults> <log_results> <WLSDPLY-05204> <Validation found 1 error, 0 warning, and 0 informational messages.>
####<Mar 28, 2019 9:47:53 AM> <SEVERE> <ValidationResults> <log_results> <WLSDPLY-05207> <See the next 1 log messages for the details.>
####<Mar 28, 2019 9:47:53 AM> <SEVERE> <ValidationResults> <log_results> <WLSDPLY-05026> <WebAppContainer is not one of the folder names allowed in model location topology:/>
####<Mar 28, 2019 9:47:53 AM> <SEVERE> <create> <validate_model> <WLSDPLY-20001> <createDomain did not create the domain because validation failed>

createDomain.sh failed (exit code = 2)
```

**Comment by rakillen** [comment]
Sorry, I should have mentioned, WebAppContainer has been added and merged into the master branch, but is not in the latest release.  We are planning release 0.21 in the next day or so, or you can build from master.

**Comment by 53d117460ec63d70** [comment]
I can confirm this is fixed in WDT 0.21, many thanks.
